components/calendar/src/buddhist.rs

```diff
@@ -220,3 +220,235 @@ fn iso_year_as_buddhist(year: i32) -> types::FormattableYear {
+        let buddhist_month = case.buddhist_month;
+        let buddhist_day = case.buddhist_day;
+
+        let iso1 = Date::try_new_iso_date(iso_year, iso_month, iso_day).unwrap();
+        let buddhist1 = iso1.to_calendar(Buddhist);
+        assert_eq!(
+            buddhist1.year().number,
+            buddhist_year,
+            "Iso -> Buddhist year check failed for case: {case:?}"
+        );
+        assert_eq!(
+            buddhist1.month().ordinal,
+            buddhist_month as u32,
+            "Iso -> Buddhist month check failed for case: {case:?}"
+        );
+        assert_eq!(
+            buddhist1.day_of_month().0,
+            buddhist_day as u32,
+            "Iso -> Buddhist day check failed for case: {case:?}"
+        );
+
+        let buddhist2 =
+            Date::try_new_buddhist_date(buddhist_year, buddhist_month, buddhist_day).unwrap();
+        let iso2 = buddhist2.to_calendar(Iso);
+        assert_eq!(
+            iso2.year().number,
+            iso_year,
+            "Buddhist -> Iso year check failed for case: {case:?}"
+        );
+        assert_eq!(
+            iso2.month().ordinal,
+            iso_month as u32,
+            "Buddhist -> Iso month check failed for case: {case:?}"
+        );
+        assert_eq!(
+            iso2.day_of_month().0,
+            iso_day as u32,
+            "Buddhist -> Iso day check failed for case: {case:?}"
+        );
+    }
+
+    #[test]
+    fn test_buddhist_cases_near_rd_zero() {
+        let cases = [
+            TestCase {
+                iso_year: -100,
+                iso_month: 2,
+                iso_day: 15,
+                buddhist_year: 443,
+                buddhist_month: 2,
+                buddhist_day: 15,
+            },
+            TestCase {
+                iso_year: -3,
+                iso_month: 10,
+                iso_day: 29,
+                buddhist_year: 540,
+                buddhist_month: 10,
+                buddhist_day: 29,
+            },
+            TestCase {
+                iso_year: 0,
+                iso_month: 12,
+                iso_day: 31,
+                buddhist_year: 543,
+                buddhist_month: 12,
+                buddhist_day: 31,
+            },
+            TestCase {
+                iso_year: 1,
+                iso_month: 1,
+                iso_day: 1,
+                buddhist_year: 544,
+                buddhist_month: 1,
+                buddhist_day: 1,
+            },
+            TestCase {
+                iso_year: 4,
+                iso_month: 2,
+                iso_day: 29,
+                buddhist_year: 547,
+                buddhist_month: 2,
+                buddhist_day: 29,
+            },
+        ];
+
+        for case in cases {
+            check_test_case(case);
+        }
+    }
+
+    #[test]
+    fn test_buddhist_cases_near_epoch() {
+        // 1 BE = 543 BCE = -542 ISO
+        let cases = [
+            TestCase {
+                iso_year: -543,
```

Comment:
Nit: Add tests for negative Buddhist years; we care about them round-tripping and having a well-defined algorithm even if it doesn't reflect a real-world date in that calendar